web/courses.go

```diff
@@ -163,7 +163,7 @@ func (s *AutograderService) getAllCourseSubmissions(request *pb.SubmissionsForCo
 	course.SetSlipDays()
 	for _, a := range assignments {
 		for _, sbm := range a.Submissions {
-			if request.GetSkipBuildInfo() {
+			if !request.GetWithBuildInfo() {
```

Comment:
Suggest passing this boolean to the database function above. That way we avoid fetching build info from the database when we don't need it. See my suggestions in issue #462 that uses the `withBuildInfo` bool; this can be passed as an argument to `GetAssignmentsWithSubmissions`... I don't think there are many uses of this method, so should be a simple change.